### ISSUE TITLE
Problem with new OpenMVG (Ubuntu 24.04)

### ISSUE THREAD

**Issue by cogitas3d** [openMVG/openMVG#2367]
Dear friends,

I'm trying to build OpenMVG for a long time and unfortunatelly something do not work anymore.

I tested with tutorial_demo.py script and, in this case works, but with my photos, that worked in old OpenMVG versions, do not works anymore.

Apparently all runs correctly until the step 4 when appear an error:

ERROR: [sequential_SfM.cpp:110] Unable to choose an initial pair, since there is no defined intrinsic data.

After this... the code stop to generate the files.

Ill let down the terminal output:

```
Optional parameters: 
--input_pairs        
--output_pairs       
--force              false
--geometric_model    f
--guided_matching    0
--cache_size         unlimited
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 50%
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 100%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 10%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 20%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 30%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 40%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 50%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 60%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 70%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 80%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 90%
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 100%
INFO: [graph_stats.hpp:53] Graph statistics:
	#nodes: 18
	#cc: 1
	#singleton: 0
	Node degree statistics:	min: 17, max: 17, mean: 17, median: 17
INFO: [main_GeometricFilter.cpp:384] Task done in (s): 0
INFO: [main_GeometricFilter.cpp:387] 
 Export Adjacency Matrix of the pairwise's geometric matches
3. Do Incremental/Sequential reconstruction
INFO: [main_SfM.cpp:135] 
-----------------------------------------------------------
 Structure from Motion:
-----------------------------------------------------------
INFO: [loggerprogress.hpp:79] [- Features Loading -] 50%
INFO: [loggerprogress.hpp:79] [- Features Loading -] 100%
INFO: [sequential_SfM.cpp:278] Track building
INFO: [sequential_SfM.cpp:281] Track filtering
INFO: [sequential_SfM.cpp:283] Track export to internal struct
INFO: [sequential_SfM.cpp:309] 
------------------
-- Tracks Stats --
 Tracks number: 15405
 Images Id: 
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 
------------------
TrackLength, Occurrence
	2	7297
	3	2926
	4	1712
	5	1043
	6	683
	7	457
	8	311
	9	219
	10	150
	11	122
	12	92
	13	77
	14	59
	15	37
	16	21
	17	42
	18	157

ERROR: [sequential_SfM.cpp:110] Unable to choose an initial pair, since there is no defined intrinsic data.
INFO: [sequential_SfM.cpp:173] Cannot find a valid initial pair - stop reconstruction.
5. Colorize Structure
ERROR: [main_ComputeSfM_DataColor.cpp:69] The input SfM_Data file "/home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/reconstruction_sequential/sfm_data.bin" cannot be read.
4. Structure from Known Poses (robust triangulation)
INFO: [main_ComputeStructureFromKnownPoses.cpp:57] Compute Structure from the provided poses
ERROR: [main_ComputeStructureFromKnownPoses.cpp:132] The input SfM_Data file "/home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/reconstruction_sequential/sfm_data.bin" cannot be read.
2. Filter matches (for the global SfM Pipeline)
INFO: [main_GeometricFilter.cpp:132]  You called : 
/home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/Linux-x86_64-RELEASE/openMVG_main_GeometricFilter
--input_file:        /home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/matches/sfm_data.json
--matches:           /home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/matches/matches.putative.bin
--output_file:       /home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/matches/matches.e.bin
Optional parameters: 
--input_pairs        
--output_pairs       
--force              false
--geometric_model    e
--guided_matching    0
--cache_size         unlimited
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 50%
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 100%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,5)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,2)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,3)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,4)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,5)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,6)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,14)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 10%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,3)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,4)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,5)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,6)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,14)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 20%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,4)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,5)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,6)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,16)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 30%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,1)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,5)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,6)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,6)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,6)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 40%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (5,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,4)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,10)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 50%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,3)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (2,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,7)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (3,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,2)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (0,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (4,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (1,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,8)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (6,17)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 60%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (7,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,9)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,10)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (8,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,11)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 70%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (9,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,11)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (10,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,12)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,14)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 80%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (11,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (12,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (12,13)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (12,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (13,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (13,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (12,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (13,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (12,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (15,16)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 90%
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (16,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (13,14)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (14,15)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (14,16)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (14,17)
WARNING: [E_ACRobust.hpp:89] Skip this pair. No intrinsic information: (15,17)
INFO: [loggerprogress.hpp:79] [- Geometric filtering -] 100%
INFO: [graph_stats.hpp:53] Graph statistics:
	#nodes: 18
	#cc: 18
	#singleton: 18
	Node degree statistics:	min: 1513409000, max: 1, mean: 0, median: 1
INFO: [main_GeometricFilter.cpp:384] Task done in (s): 0
INFO: [main_GeometricFilter.cpp:387] 
 Export Adjacency Matrix of the pairwise's geometric matches
3. Do Global reconstruction
INFO: [main_SfM.cpp:135] 
-----------------------------------------------------------
 Structure from Motion:
-----------------------------------------------------------
INFO: [loggerprogress.hpp:79] [- Features Loading -] 50%
INFO: [loggerprogress.hpp:79] [- Features Loading -] 100%
INFO: [connectedComponent.hpp:99] CleanGraph_KeepLargestBiEdge_Nodes():: => connected Component: 0
WARNING: [sfm_global_engine_relative_motions.cpp:122] Invalid input image graph for global SfM
5. Colorize Structure
ERROR: [main_ComputeSfM_DataColor.cpp:69] The input SfM_Data file "/home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/reconstruction_global/sfm_data.bin" cannot be read.
4. Structure from Known Poses (robust triangulation)
INFO: [main_ComputeStructureFromKnownPoses.cpp:57] Compute Structure from the provided poses
ERROR: [main_ComputeStructureFromKnownPoses.cpp:132] The input SfM_Data file "/home/cogitas3d/Programs/Fotogrametria/openMVG_aruco/software/SfM/tutorial_out/reconstruction_global/sfm_data.bin" cannot be read.

```

**Comment by pmoulon** [comment]
It must be the way we are estimating the camera focal length that does not work for your camera. You can try specify it with -f parameter on the sfm unit step manually. A value like 1.2 * max(img width, height) should do it as a starting point.

**Comment by cogitas3d** [comment]
Dear @pmoulon 

I did it and works, thank you! But... 
I export to OpenMVS and crash there before starts to reconstruc the scene, or... reconstruc the dense cloud and after, when is finishing the mesh reconstruction... crash.
I dont know if the problem is on MVS or on the OpenMVG exporter.

**Comment by cogitas3d** [comment]
@pmoulon I will report on the progress of the studies. After I introduced the argument you observed, all calculations were performed, then I processed them in openMVS with that erros I repored you. Initially I thought it was purely a compilation error or library incompatibility, but I realized that the dense cloud of points is much denser than in the previous version of the program, so dense that I could not even see the points when I got as close as possible in Meshlab. So, I started increasing the value of the -d argument and the mesh was reconstructed, so, curiously, we may be facing "a good problem", that is, the data is so dense that it causes the program to collapse. This lip image you are seeing in the attachment is a cloud of points... not a mesh and the camera is very close, without the points being visible. 

![Captura de tela de 2024-10-21 17-28-04](https://github.com/user-attachments/assets/58445ae1-cd1f-475d-909b-272b6bf890e2)

However, I also noticed a problem, I ran a code where I had compared OpenMVG+OpenMVS with other photogrammetry programs (https://ortogonline.com/doc/pt_br/OrtogOnLineMag/5/Fotogrametria.html) and, this time it generated a mesh with more details than that time, but it ended up generating many more points, which are not part of the scene and this made the model very heavy. See the attached image and notice that the extra points ended up being reconstructed (I had to reconstruct it in Meshlab, because it crashed in MVS).

![Captura de tela de 2024-10-22 07-44-05](https://github.com/user-attachments/assets/a4209eaa-a7a7-477a-bace-81c94136cfc5)
